### PR TITLE
Add Support for PC311TY Energy Meter

### PR DESCRIPTION
**Pull request by Doudou14**
Add pc311ty_energy_meter.

### CODE REVIEW

**Comment by Doudou14** [comment]
Update pc311ty_energy_meter.yaml
Add missing entity
Fix missing class Energy sensor
Fix "Generation B"

File: custom_components/tuya_local/devices/pc311ty_energy_meter.yaml [hunk 1]
```diff
@@ -0,0 +1,222 @@
+name: PC311-TY power clamp
+products:
+  - id: vfwxetkavqrvg02u
+primary_entity:
+  entity: sensor
+  category: diagnostic
+  class: power
+  name: Total active power
```
Comment (make-all) [inline custom_components/tuya_local/devices/pc311ty_energy_meter.yaml:8]: For an energy meter, would it make more sense for the Total Energy reading to be the primary sensor?

**Comment by make-all** [comment]
Please separate the translations into a separate PR

File: custom_components/tuya_local/devices/pc311ty_energy_meter.yaml (hunk 1 above)
Comment (Doudou14) [reply custom_components/tuya_local/devices/pc311ty_energy_meter.yaml]: I changed the primary sensor

File: custom_components/tuya_local/devices/pc311ty_energy_meter.yaml [hunk 2]
```diff
@@ -2,18 +2,6 @@ name: PC311-TY power clamp
...
-  name: Total active power
-  dps:
-    - id: 9
-      type: integer
-      name: sensor
-      unit: W
-      force: true
-      class: measurement
-secondary_entities:
   - entity: sensor
```
Comment (make-all) [inline custom_components/tuya_local/devices/pc311ty_energy_meter.yaml:5]: You'll need to remove the `-` here, as primary entity is not a list.

**Review by make-all — CHANGES_REQUESTED** [review]

File: custom_components/tuya_local/devices/pc311ty_energy_meter.yaml (hunk 2 above)
Comment (Doudou14) [reply custom_components/tuya_local/devices/pc311ty_energy_meter.yaml]: I think this is good, sorry I am beginner to home assistant

**Review by make-all — APPROVED** [review]

**Comment by make-all** [comment]
Changes have been merged

**Pull request closed without merging**